In [5]:
import numpy as np
import scipy.stats as stats
from math import *

#### Задание 1<br>Известно, что генеральная совокупность распределена нормально со средним квадратическим отклонением, равным 16.\Найти доверительный интервал для оценки математического ожидания a с надежностью 0.95, если выборочная средняя M = 80, а объем выборки n = 256.

$ \bar X \pm z_{\alpha/2}\cdot\frac{\sigma}{\sqrt n}$<br>
$Z_{\alpha/2}$ по таблице = 1.96

In [3]:
left=80-1.96*16/256**0.5
right=80+1.96*16/256**0.5
print(f'[{left} : {right}]')

[78.04 : 81.96]


Ответ: Доверительный интервал равен [78.04 : 81.96]

#### Задание 2<br> В результате 10 независимых измерений некоторой величины X, выполненных с одинаковой точностью,  получены опытные данные: 6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1  Предполагая, что результаты измерений подчинены нормальному закону распределения вероятностей, оценить истинное значение величины X  при помощи доверительного интервала, покрывающего это значение с доверительной вероятностью 0,95.

$ \displaystyle\bar X \pm t_{\alpha/2}\cdot\frac{\sigma}{\sqrt n}$

In [13]:
ar = np.array([6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1])
ta = stats.t.ppf(0.975, len(ar)-1) #t критерий для a/2
sigma = np.std(ar, ddof=1) # сигма выборки
x = ar.mean() # мат. ожидание
left = x - ta*sigma/np.sqrt(len(ar))
right = x + ta*sigma/np.sqrt(len(ar))
left, right

(6.267515851415713, 6.912484148584288)

Ответ: истинное значение находится в доверительном интервале<br> [6.267515851415713 : 6.912484148584288]

#### Задание 3<br>Рост дочерей 175, 167, 154, 174, 178, 148, 160, 167, 169, 170  Рост матерей 178, 165, 165, 173, 168, 155, 160, 164, 178, 175  Используя эти данные построить 95% доверительный интервал для разности среднего роста родителей и детей.

$\bar{X}_1 - \bar{X}_2 \pm t_{\alpha/2} \sqrt{\frac{s_1^2}{n_1} + \frac{s_2^2}{n_2}}$

In [15]:
mot = np.array([178, 165, 165, 173, 168, 155, 160, 164, 178, 175])
dau = np.array([175, 167, 154, 174, 178, 148, 160, 167, 169, 170])
dif = mot.mean() - dau.mean() # разность средних
se = np.sqrt(np.var(mot, ddof=1)/len(mot) + np.var(dau, ddof=1)/len(dau)) # Ошибка разности
t = stats.t.ppf(0.975, len(mot) + len(dau) - 2) # t критерий 
left = dif - t * se
right = dif + t * se
left, right

(-6.268418034506846, 10.068418034506857)

Ответ: доверительный интервал для разности среднего роста родителей и детей<br> [-6.268418034506846, 10.068418034506857]